### PR TITLE
Port sriov interface to ppx

### PR DESCRIPTION
**Pull request by minli1**
This is for after rebase master then changes the interface to PPX.
Signed-off-by: Yang Qian <yang.qian@citrix.com>

### CODE REVIEW

File: lib/xcp_pci.ml [hunk 1]
```diff
@@ -6,7 +6,7 @@ type address = {
 	dev: int;
 	fn: int;
 }
-[@@deriving sexp, rpc]
+[@@deriving sexp, rpc, rpcty]
```
Comment (mseri) [inline lib/xcp_pci.ml]: Do we use `rpc`? Can we just use `rpcty` instead, as done in other modules or itwill break lots of other code?

File: network/network_interface.ml [hunk 2]
```diff
@@ -610,39 +614,63 @@ module Interface_API(R : RPC) = struct
...
+      let result_p = Param.mk ~description:["SR-IOV disable result"] disable_result in
+      declare
+        "Sriov.disable"
+        ["Disable SR-IOV"]
+        (debug_info_p @-> iface_name_p @-> returning result_p err)
+
+    let make_vf_config =
+      let pci_address_p = Param.mk ~description:["pci address"] Xcp_pci.address in
+      let vf_info_p = Param.mk ~description:["vf info"] sriov_pci_t in
+      let result_t = Param.mk ~description:["SR-IOV make vf configuration result"] config_result in
```
Comment (mseri) [inline network/network_interface.ml]: Why not `result_p` in line with all the other declarations?

File: lib/xcp_pci.ml (hunk 1 above)
Comment (robhoes) [reply lib/xcp_pci.ml]: I think that this is shared with the xenops_interface, which is still on camlp4.

Comment (mseri) [reply lib/xcp_pci.ml]: Non of the three things above is related to camlp4 though. I am expecting that removing rpc fails something, but I think it's worth double checking and see if we can reduce the autogenerated stuff if not needed

Comment (robhoes) [reply lib/xcp_pci.ml]: Without it, xcpi-idl fails to build:
```
File "xen/xenops_types.ml", line 19, characters 2-224:
Error: Unbound value rpc_of_address
Hint: Did you mean typ_of_address?
make: *** [Makefile:7: release] Error 1
```

Comment (mseri) [reply lib/xcp_pci.ml]: Good, let's merge then

**Pull request merged**